pkg/helm/informers.go

```diff
@@ -0,0 +1,179 @@
+package helm
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/mitchellh/hashstructure"
+	"github.com/pkg/errors"
+	apptypes "github.com/replicatedhq/kots/pkg/app/types"
+	"github.com/replicatedhq/kots/pkg/appstate"
+	appstatetypes "github.com/replicatedhq/kots/pkg/appstate/types"
+	identitydeploy "github.com/replicatedhq/kots/pkg/identity/deploy"
+	identitytypes "github.com/replicatedhq/kots/pkg/identity/types"
+	"github.com/replicatedhq/kots/pkg/kotsutil"
+	"github.com/replicatedhq/kots/pkg/logger"
+	registrytypes "github.com/replicatedhq/kots/pkg/registry/types"
+	"github.com/replicatedhq/kots/pkg/render"
+	"github.com/replicatedhq/kots/pkg/template"
+	"github.com/replicatedhq/kots/pkg/util"
+	"k8s.io/client-go/kubernetes"
+)
+
+var monitorMap map[string]*appstate.Monitor
+
+func initMonitor(clientset kubernetes.Interface, targetNamespace string) {
+	if monitorMap == nil {
+		monitorMap = make(map[string]*appstate.Monitor)
+	}
+
+	if monitorMap[targetNamespace] == nil {
+		monitorMap[targetNamespace] = appstate.NewMonitor(clientset, targetNamespace)
+	}
+
+	go runAppStateMonitor(monitorMap[targetNamespace])
+}
+
+func resumeHelmStatusInformers() {
+	for _, appName := range GetCachedHelmApps() {
+		helmApp := GetHelmApp(appName)
+		app := &apptypes.App{
+			ID:                helmApp.GetID(),
+			Slug:              helmApp.GetSlug(),
+			Name:              helmApp.Release.Name,
+			IsAirgap:          helmApp.GetIsAirgap(),
+			CurrentSequence:   helmApp.GetCurrentSequence(),
+			UpdatedAt:         &helmApp.CreationTimestamp,
+			CreatedAt:         helmApp.CreationTimestamp,
+			LastUpdateCheckAt: &helmApp.CreationTimestamp,
+			IsConfigurable:    helmApp.IsConfigurable,
+		}
+
+		kotsKinds, err := GetKotsKindsFromHelmApp(helmApp)
+		if err != nil {
+			logger.Errorf("failed to get kots kinds from helm app: %v\n", err)
+			return
+		}
+
+		settings := registrytypes.RegistrySettings{
+			Namespace: util.PodNamespace,
+		}
+		builder, err := render.NewBuilder(&kotsKinds, settings, app.Slug, helmApp.GetCurrentSequence(), app.IsAirgap, util.PodNamespace)
+		if err != nil {
+			logger.Errorf("failed to create new builder: %v\n", err)
+			return
+		}
+
+		// apply the informers for this app since they havent been applied yet
```

Comment:
Is this true?  This function is called every time a secret is added or removed, and it looks at every app in the cache every time